flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1848,3 +2143,310 @@ def test_run_mkfs_and_mount(self):
                 in psutil.disk_partitions()
             )
         )
+
+
+class ResizeBlockDeviceDatasetInitTests(
+    make_with_init_tests(
+        ResizeBlockDeviceDataset,
+        dict(dataset_id=uuid4(), size=REALISTIC_BLOCKDEVICE_SIZE),
+        dict(),
+    )
+):
+    """
+    Tests for ``ResizeBlockDeviceDataset`` initialization.
+    """
+
+
+class ResizeBlockDeviceDatasetTests(
+    make_istatechange_tests(
+        ResizeBlockDeviceDataset,
+        lambda _uuid=uuid4(): dict(
+            dataset_id=_uuid, size=REALISTIC_BLOCKDEVICE_SIZE
+        ),
+        lambda _uuid=uuid4(): dict(
+            dataset_id=_uuid, size=REALISTIC_BLOCKDEVICE_SIZE
+        ),
+    )
+):
+    """
+    Tests for ``ResizeBlockDeviceDataset``.
+    """
+    def test_dataset_id_required(self):
```

Comment:
Yea... If I were coming at it now, I might not write them.  Or I might factor this kind of testing into the `make_with_init_tests` helper (or a similar new one).  Going to just leave these for now though; they're not **in**correct.